File: Data_structures/circularQueue.c
#include<stdio.h>
#include<stdlib.h>

struct node
{
	int data;
	struct node* next;
};
struct node *front = NULL;
struct node *rear = NULL;

void enqueue(int d) //Insert elements in Queue
{
	struct node* n;
	n = (struct node*)malloc(sizeof(struct node));
	n->data = d;
	n->next = NULL;
	if((rear==NULL)&&(front==NULL))
	{
		front = rear = n;
		rear->next = front;
	}
	else
	{
		rear->next = n;
		rear = n;
		n->next = front;
	}
}

void dequeue() // Delete an element from Queue
{
	struct node* t;
	t = front;
	if((front==NULL)&&(rear==NULL))
		printf("\nQueue is Empty");
	else if(front == rear){
		front = rear = NULL;
		free(t);
	}
	else{
		front = front->next;
		rear->next = front;
		free(t);
  }	
}

void print(){ // Print the elements of Queue
	struct node* t;
	t = front;
	if((front==NULL)&&(rear==NULL))
		printf("\nQueue is Empty");
	else{
		do{
			printf("\n%d",t->data);
			t = t->next;
		}while(t != front);
	}
}

int main()
{
	int choice,n,i,data;
	printf("Enter Your Choice:-");
	do{
		printf("\n\n1. Insert the Data in Queue\n2. Display the Data in Queue \n3. Delete the data from the Queue\n0. Exit\n");
		scanf("%d",&choice);
		switch(choice){
			case 1:
				printf("\nEnter the number of data: ");
				scanf("%d",&n);
				printf("\nEnter your data\n");
				i=0;
				while(i<n){
					scanf("%d",&data);
					enqueue(data);
					i++;
				}
				break;
			case 2:
				print();
				break;
			case 3:
				 dequeue();
				break;
			case 0:
				break;
			default:
				printf("\nIncorrect Choice");	
		}
	}while(choice!=0);
return 0;
}

File: Data_structures/CircularQueue.py
# Code by : https://github.com/MaxTronn

# This program creates a Circular Queue and displays it
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class CircularQueue:
    def __init__(self):
        self.front = None
        self.rear = None

# This function adds elements to the circular queue
def enqueue(queue, item):
    tempnode = Node(item)
    tempnode.data = item

    if queue.front is None:
        queue.front = tempnode
    else:
        queue.rear.next = tempnode

    queue.rear = tempnode
    queue.rear.next = queue.front

# This function prints elements of the circular queue
def printQueue(queue):
    tempnode = queue.front
    print("Elements in Circular Queue are: ",
                                   end = " ")
    while tempnode.next != queue.front:
        print(tempnode.data, end = " ")
        tempnode = tempnode.next
    print(tempnode.data)


#Driver Code
queue = CircularQueue()

enqueue(queue, 20)
enqueue(queue, 45)
enqueue(queue, 77)
enqueue(queue, 100)

printQueue(queue)